### DISCUSSION TITLE
Language Freeze for 3.9, 3.10 and 4.0 until stable release of 4.0

### DISCUSSION THREAD

**Discussion by wilsonge** [joomla/joomla-cms#34685] [Show and tell]
As an FYI for all those contributing we will call language freeze on all new strings immediately after tagging 4.0 RC4. This is to give Translation teams maximum time to be able to translate their language packs.

All PR’s after RC4 that have a language string change will be given a target of 4.0.1

**Comment by brianteeman** [comment]
Just to be clear. Are you referring to just brand new strings that never existed before or to changes to existing strings as well

**Reply by wilsonge** [reply]
Both new and existing. No changes to strings outright. This gives time (there'll be at minimum 1 month - minimum 2 more RC's after RC4) for any TT's to start releasing stable packs ahead of a stable release.

Obviously in some exceptional circumstances I may merge PRs (security issues/critical issues that require a new string) but the general assumption should be is they'll be left for future releases. From 4.0.1 onwards we'll follow the usual 3.x process on language freeze just a week ahead. But giving longer for this release as TT's will need to create full packs.

**Reply by brianteeman** [reply]
Thanks for clarifying as the first sentence said **new strings** and the third sentence said **string change**

**Reply by richard67** [reply]
How is it with changes on existing strings which only change capitalization e.g. in titles, buttons or labels? Those would not really require new translations since in other languages there are different capitalization rules anyway for that.

**Comment by Kostelano** [comment]
@wilsonge By the way, when can we include the installation language files in the Joomla 4.0 distribution? I think that many have them ready.

**Reply by HLeithner** [reply]
@Kostelano installation language can already be translated on joomla.crowdin.com the new system will automatically push the translations https://github.com/joomla/core-translations/tree/main/joomla_v4/translations/core/installation/language which will automatically create a PR against the cms repo. (@Llewellynvdm is volunteering for the PR script)

**Comment by infograf768** [comment]
Don’ forget to inform tts by the private translation forum and the private mailing list

**Reply by zero-24** [reply]
I have just posted to that forum with the note and a link over here.

**Comment by HLeithner** [comment]
We decided to have a language freeze from today till Joomla! 4.0 is finally released for the branches 3.9, 3.10 and 4.0.

(there maybe be some exceptions but they will be decided by release leads)
